server/consumer.go

```diff
@@ -1830,6 +1834,27 @@ func (o *consumer) writeStoreStateUnlocked() error {
 	return o.store.Update(&state)
 }
 
+// Returns an initial info. Only applicable for non-clustered consumers.
+// We will clear after we return it, so one shot.
+func (o *consumer) initialInfo() *ConsumerInfo {
+	o.mu.Lock()
+	ici := o.ici
+	o.ici = nil // gc friendly
+	o.mu.Unlock()
+	if ici == nil {
+		ici = o.info()
```

Comment:
If we set o.ici inside o.info() *and* we would not want it to be set in that specific condition, then maybe we need a version of o.info() (since this is called many times) where we pass a boolean to indicate if we want to set it or not. So for instance, in setLeader() we would pass true so that it is set, but everywhere else it would not.